dependency/vault_write_test.go

```diff
@@ -71,6 +71,69 @@ func TestNewVaultWriteQuery(t *testing.T) {
 	}
 }
 
+func TestVaultWriteSecretKV_Fetch(t *testing.T) {
+	t.Parallel()
+
+	// trigger a nil-pointer-deref panic in wq.Fetch() with KVv1 due to
+	// writeSecret() returning nil for vaultSecret
+	t.Run("write_secret_v1", func(t *testing.T) {
+		clients, vault := testVaultServer(t, "write_secret_v1", "1")
+		secretsPath := vault.secretsPath
+
+		path := secretsPath + "/foo"
+		exp := map[string]interface{}{
+			"bar": "zed",
+		}
+
+		wq, err := NewVaultWriteQuery(path, exp)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		_, _, err = wq.Fetch(clients, &QueryOptions{})
+		if err != nil {
+			fmt.Println(err)
+		}
+
+		rq, err := NewVaultReadQuery(path)
```

Comment:
should this fail on err?

Reply:
Sure should. Seems like the linter should pick out unhandled err values like that. I overlooked it as I was focused on using it to test the value read, not so much the reader.